src/main/java/org/mastodon/mamut/tomancak/MirrorEmbryo.java

```diff
@@ -0,0 +1,97 @@
+package org.mastodon.mamut.tomancak;
+
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import javax.swing.JOptionPane;
+
+import org.mastodon.mamut.MamutAppModel;
+import org.mastodon.mamut.model.Model;
+import org.mastodon.mamut.model.ModelGraph;
+import org.mastodon.mamut.model.Spot;
+
+/**
+ * Plugin that mirrors the X coordinate of each spot.
+ */
+public class MirrorEmbryo
+{
+
+	/**
+	 * Shows a warning dialog and mirrors all spots along the X axis.
+	 */
+	public static void run( MamutAppModel appModel )
+	{
+		if ( !showDialog() )
+			return;
+		Model model = appModel.getModel();
+		ModelGraph graph = model.getGraph();
+		ReentrantReadWriteLock.WriteLock lock = graph.getLock().writeLock();
+		lock.lock();
+		try
+		{
+			mirrorX( graph );
+			model.setUndoPoint();
+		}
+		finally
+		{
+			lock.unlock();
+		}
+	}
+
+	private static boolean showDialog()
+	{
+		String title = "Mirror X coordinate of each spot";
+		String message = "Mirror the X coordinate of each spot.\n"
+				+ "\n"
+				+ "The plugin first calculates the mean X coordinate of all spots."
+				+ "Then the X-coordinate of each spot is mirrored around the"
+				+ "plane x = mean X. The ellipsoids are mirrored as well.\n"
+				+ "\n"
+				+ "Please note: The plugin does not support mirroring of the image data.\n"
+				+ "The spots will therefore appear to be in the wrong place.\n"
+				+ "\n"
+				+ "Do you want to continue?";
+		String[] buttons = { "Mirror", "Cancel" };
+		String defaultButton = "Mirror";
+		int result = JOptionPane.showOptionDialog( null, message, title, JOptionPane.YES_NO_OPTION,
```

Comment:
This creates a modal dialog. I think it would be better for user experience, if it was non-modal. Also, it would be more consistent with dialogs from the same menu, e.g. "Sort lineage tree" opens a non-modal dialog

Reply:
I disagree. I think the modal dialog is the best option for this simple yes-no-qestion. But I understand that the non-modal dialogs are the better option for more complex dialogs, where the users might like to switch windows to look up information they would need to enter. But this dialog does not ask for any information, it's a simple warning. And the model dialog seems to be best suited for this.